File: menu-scanner-backend/src/main/java/com/emenu/features/order/dto/response/PaymentOptionResponse.java
package com.emenu.features.order.dto.response;

import com.emenu.enums.common.Status;
import com.emenu.enums.payment.PaymentOptionType;
import com.emenu.shared.dto.ImageUrls;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PaymentOptionResponse {

    private UUID id;
    private UUID businessId;
    private String name;
    private String description;
    private PaymentOptionType paymentOptionType;
    private Status status;
    private ImageUrls image;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: menu-scanner-frontend-owner/src/features/master-data/store/models/response/payment-option-response.ts
import { Status } from "@/constants/status/status";
import { BasePagination } from "@/utils/common/pagination";
import { ImageUrls } from "@/features/auth/store/models/request/users-request";

export type PaymentOptionType = "CASH";

export interface AllPaymentOptionResponseModel extends BasePagination {
  content: PaymentOptionResponse[];
}

export interface PaymentOptionResponse {
  id: string;
  businessId: string;
  name: string;
  description?: string;
  paymentOptionType: PaymentOptionType;
  status: Status;
  image?: ImageUrls;
  createdAt: string;
  updatedAt: string;
}
